homeassistant/components/backup/http.py

```diff
@@ -58,12 +65,25 @@ async def get(
         headers = {
             CONTENT_DISPOSITION: f"attachment; filename={slugify(backup.name)}.tar"
         }
+        reader: IO[bytes]
         if agent_id in manager.local_backup_agents:
             local_agent = manager.local_backup_agents[agent_id]
             path = local_agent.get_backup_path(backup_id)
-            return FileResponse(path=path.as_posix(), headers=headers)
-
-        stream = await agent.async_download_backup(backup_id)
+            if not password:
+                return FileResponse(path=path.as_posix(), headers=headers)
+            try:
+                reader = await hass.async_add_executor_job(open, path.as_posix(), "rb")
+            except FileNotFoundError:
+                return Response(status=HTTPStatus.NOT_FOUND)
+        else:
+            stream = await agent.async_download_backup(backup_id)
+            if password:
+                reader = cast(IO[bytes], util.AsyncIteratorReader(hass, stream))
+        if password:
+            stream = util.AsyncIteratorWriter(hass)
+            threading.Thread(
```

Comment:
Should we ensure the thread stops?

Reply:
When stopping Home Assistant?

Reply:
At that point it happens automatically. But I think we might want to wait for the thread to stop when finishing the http request, to ensure we don't leak threads.

Reply:
Right, this thread is not intended as a background thread but just a way to not tie up an executor thread.

If we can use an async synchronization it sounds good to wait.

Reply:
I added a mechanism so we can wait for the worker to complete